rules/mockery-to-prophecy/src/Rector/MethodCall/MockeryPredictionsToProphizeRector.php

```diff
@@ -0,0 +1,220 @@
+//            }
+//
+////             if ($this->isName($node, 'never')) {
+////                return $this->createMethodCall(
+////                    $node->var,
+////                    'shouldNotBeCalled'
+////                );
+////            }
+//        }
+
+        return $node;
+    }
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition(
+            'Removes mockery close from test classes',
+            [
+                new CodeSample(
+                    '$mock->shouldRecieve("someMethod")->never()',
+                    '$mock->someMethod()->shouldNotBeCalled()'
+                ),
+            ]
+        );
+    }
+
+    private function replaceMockCreationsAndCollectVariableNames(ClassMethod $node)
+    {
+        $this->traverseNodesWithCallable($node->stmts, function (Node $node) {
+            if (!$this->isStaticCallNamed($node, 'Mockery', 'mock')) {
+                return null;
+            }
+
+            $this->collectMockVariableName($node);
+
+            if ($node->getAttribute('parentNode') instanceof Node\Arg) {
+                return $this->createMethodCall($this->prophesizeCreateMock($node), 'reveal');
+            }
+
+            return $this->prophesizeCreateMock($node);
+        });
+    }
+
+    private function collectMockVariableName(Node\Expr\StaticCall $staticCall): void
+    {
+        $parentNode = $staticCall->getAttribute(AttributeKey::PARENT_NODE);
+        if (! $parentNode instanceof Assign) {
+            return;
+        }
+
+        if (! $parentNode->var instanceof Variable) {
+            return;
+        }
+
+        /** @var Variable $variable */
+        $variable = $parentNode->var;
+
+        /** @var string $variableName */
+        $variableName = $this->getName($variable);
+
+        $type = $staticCall->args[0]->value;
+        $mockedType = $this->getValue($type);
+
+        $this->mockVariableTypesByNames[$variableName] = $mockedType;
+    }
+
+
+    /**
+     * @param $node
+     * @return MethodCall
+     */
+    private function prophesizeCreateMock($node): MethodCall
+    {
+        return $this->createLocalMethodCall('prophesize', [$node->args[0]]);
+    }
+
+    private function replaceMockExpectations(ClassMethod $node)
+    {
+        $this->traverseNodesWithCallable($node->stmts, function (Node $node) {
+            if (!$this->isMethodCallOnMockVariable($node)) {
+                return null;
+            }
+
+            if ($node instanceof MethodCall) {
+                /** @var $node MethodCall */
+                if ($this->isName($node->name, 'shouldReceive')) {
+                    if (!isset($node->args[0]) || !$node->args[0]->value instanceof String_) {
+                        throw new ShouldNotHappenException();
+                    }
+
+                    $methodCall = new MethodCall($node->var, $node->args[0]->value->value);
+
+                    $nextCall = $node->getAttribute('nextNode');
```

Comment:
@TomasVotruba methods there are chained. But I could not really find how to rewrite the chain properly. This is just a first test case. using `$mock->shouldReceive('create')->never();` to `$mock->create()->shouldNotBeCalled();` 

I found that `methodCall of `shouldReceive` has a nextNode of the type `Identifier` Like I wrote here. But the current result is that `create` is appended with `shouldNotBeCalled` like I want it to be. But `never` persists. So that's not removed from the tree somehow?

Reply:
Use this service:
https://github.com/rectorphp/rector/blob/71247e15877c043fbd2a251e16eb299576b23b2c/rules/magic-disclosure/src/NodeAnalyzer/ChainMethodCallNodeAnalyzer.php

Reply:
Here is how it's used in one rule: https://github.com/rectorphp/rector/blob/master/rules/magic-disclosure/src/Rector/MethodCall/DefluentMethodCallRector.php#L108-L112